cc_core.c

```diff
@@ -3677,9 +3677,85 @@ int global_static_array(struct type* type_size, char* name)
 
 	/* Ensure properly closed */
 	require_match("missing close bracket\n", "]");
-	require_match("missing ;\n", ";");
 
-	global_variable_zero_initialize(size);
+	if(global_token->s[0] == '=')
+	{
+		global_token = global_token->next;
+		require(global_token != NULL, "NULL token received in global list initializer");
+
+		/* Default to size == 4 */
+		char* prefix = "%";
+		if(type_size->size == 2)
+		{
+			prefix = "@";
```

Comment:
I don't think that is going to work with RISC-V word style prefixes.

(And eventually that style might get used with other arches where that makes sense, I think @oriansj wants to use it for ppc64, and it might also make sense in aarch64.)

I'll try to test it on RISC-V but I think this will fail.

Reply:
I tested the newest commit out on RISCV64 and it fails. I'll redo it to output 32 bit values with `%` instead and see if that fixes it.

Reply:
Changed it to output 32 bit values instead. This seems to pass on RISCV32 and RISCV64.

It will necessarily pad values to 4 bytes but I don't think that's a bad idea.

Reply:
Thanks, I still need to properly review last 3 commits or so but I guess it will be fine now.